app/src/main/java/com/iron/espresso/presentation/home/study/SearchPlaceActivity.kt

```diff
@@ -0,0 +1,146 @@
+package com.iron.espresso.presentation.home.study
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.MenuItem
+import android.view.ViewGroup
+import android.view.inputmethod.EditorInfo
+import android.widget.EditText
+import android.widget.LinearLayout
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.DividerItemDecoration
+import com.iron.espresso.R
+import com.iron.espresso.ToolbarHelper
+import com.iron.espresso.databinding.ActivitySearchPlaceBinding
+import com.iron.espresso.model.api.KakaoApi
+import com.iron.espresso.model.response.Place
+import com.iron.espresso.model.response.PlaceResponse
+import com.iron.espresso.presentation.home.study.adapter.PlaceAdapter
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+class SearchPlaceActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivitySearchPlaceBinding
+    private lateinit var toolbarHelper: ToolbarHelper
+    private lateinit var searchEditText: EditText
+    private val placeAdapter = PlaceAdapter()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_search_place)
+
+        searchEditText = EditText(this)
+        toolbarHelper = ToolbarHelper(this, binding.appbar).apply {
+            setTitle(null)
+            setNavigationIcon(R.drawable.ic_back_24)
+            setCustomView(searchEditText.apply {
+                hint = TOOLBAR_HINT
+                setSingleLine()
+                layoutParams = ViewGroup.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT
+                )
+                requestFocus()
+                imeOptions = EditorInfo.IME_ACTION_SEARCH
+                setOnEditorActionListener { search, action, event ->
+                    var handled = false     //키보드 내림
+                    if (search.text.isNotEmpty()) {
+                        searchPlace(text.toString())
+                        Log.d("TAG", "성공")
+                    } else {
+                        Log.d("TAG", "실패")
+                        handled = true      //키보드 유지
+                    }
+                    handled
+                }
+            })
+        }
```

Comment:
```suggestion
        searchEditText = EditText(this).apply {
                hint = TOOLBAR_HINT
                setSingleLine()
                layoutParams = ViewGroup.LayoutParams(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    ViewGroup.LayoutParams.MATCH_PARENT
                )
                requestFocus()
                imeOptions = EditorInfo.IME_ACTION_SEARCH
                setOnEditorActionListener { search, action, event ->
                    var handled = false     //키보드 내림
                    if (search.text.isNotEmpty()) {
                        searchPlace(text.toString())
                        Log.d("TAG", "성공")
                    } else {
                        Log.d("TAG", "실패")
                        handled = true      //키보드 유지
                    }
                    handled
                }
            }
        toolbarHelper = ToolbarHelper(this, binding.appbar).apply {
            setTitle(null)
            setNavigationIcon(R.drawable.ic_back_24)
            setCustomView(searchEditText)
        }
```
이런식으로 정리하는게 알아보기 편할 것 같아요

Reply:
수정했습니다. 앞으로 신경쓰면서 코드 짜볼게요